boa_engine/src/vm/call_frame/mod.rs

```diff
@@ -42,7 +43,12 @@ pub struct CallFrame {
     /// \[\[ScriptOrModule\]\]
     pub(crate) active_runnable: Option<ActiveRunnable>,
 
-    pub(crate) active_function: Option<JsObject>,
+    pub(crate) environments: EnvironmentStack,
+
+    pub(crate) realm: Realm,
+
+    pub(crate) exit_early: bool,
```

Comment:
Thought: Would it be better to have a singular `bitflag` with more states eventually over a `exit_early` and `construct`?

I'd always hoped to eventually return to `CompletionType` and remove `CompletionType::Yield` in favor of the spec's `SuspendedYield` state. This may be a bit out of scope on this PR, but I thought I'd bring it up on the off chance you had run across something while working on this or may have had some thoughts about it.